### PR TITLE
HA-24 Add GithubWebhook.

### PR DESCRIPTION
**Pull request by weeellz**

### CODE REVIEW

File: HaroldAdviser/Controllers/PipelineController.cs [hunk 1]
```diff
@@ -1,52 +1,52 @@
...
+
+        public PipelineController(IPipelineManager pipelineManager)
+        {
+            _pipelineManager = pipelineManager;
+        }
+
+        [HttpPost, Route("Api/Pipeline/Create")]
+        public async Task<IActionResult> CreatePipeline([FromBody] GithubWebhook webhook)
+        {
+            //TODO: handle github webhook to create pipeline
```
Comment (fexolm) [inline HaroldAdviser/Controllers/PipelineController.cs]: Please remove comment

File: HaroldAdviser/Controllers/UserController.cs [hunk 2]
```diff
@@ -113,5 +115,17 @@ public IActionResult ShowRepositories()
             await _context.SaveChangesAsync();
             return Ok();
         }
+
+        [HttpGet, Authorize, Route("/Api/User/Repository/Warnings")]
+        public IActionResult GetWarnings()
```
Comment (fexolm) [inline HaroldAdviser/Controllers/UserController.cs:122]: Warnings are warnings, not strings.

Comment (fexolm) [reply HaroldAdviser/Controllers/UserController.cs:122]: Please add at least 6 warnings with long strings. You could use that json to get them.
``` [/home/fexolm/git/SLama/src/block.c:13]: (portability) 'mem' is of type 'void *'. When using void pointers in calculations, the behaviour is undefined.
[/home/fexolm/git/SLama/tests/block-test.c:15] -> [/home/fexolm/git/SLama/tests/block-test.c:16]: (style) Variable 'block' is reassigned a value before the old one has been used.
[/home/fexolm/git/SLama/tests/block-test.c:30] -> [/home/fexolm/git/SLama/tests/block-test.c:33]: (style) Variable 'block' is reassigned a value before the old one has been used.
[/home/fexolm/git/SLama/tests/block-test.c:31] -> [/home/fexolm/git/SLama/tests/block-test.c:35]: (style) Variable 'block2' is reassigned a value before the old one has been used.
[/home/fexolm/git/SLama/tests/block-test.c:55] -> [/home/fexolm/git/SLama/tests/block-test.c:58]: (style) Variable 'block' is reassigned a value before the old one has been used.
[/home/fexolm/git/SLama/tests/block-test.c:79] -> [/home/fexolm/git/SLama/tests/block-test.c:82]: (style) Variable 'block' is reassigned a value before the old one has been used.
[/home/fexolm/git/SLama/tests/block-test.c:80] -> [/home/fexolm/git/SLama/tests/block-test.c:83]: (style) Variable 'block2' is reassigned a value before the old one has been used.
[/home/fexolm/git/SLama/tests/matrix-base-algo-test.c:16] -> [/home/fexolm/git/SLama/tests/matrix-base-algo-test.c:20]: (style) Variable 'matrix' is reassigned a value before the old one has been used.
[/home/fexolm/git/SLama/tests/matrix-base-algo-test.c:17] -> [/home/fexolm/git/SLama/tests/matrix-base-algo-test.c:21]: (style) Variable 'matrix1' is reassigned a value before the old one has been used.
[/home/fexolm/git/SLama/tests/matrix-base-algo-test.c:51] -> [/home/fexolm/git/SLama/tests/matrix-base-algo-test.c:55]: (style) Variable 'matrix' is reassigned a value before the old one has been used.
[/home/fexolm/git/SLama/tests/matrix-base-algo-test.c:52] -> [/home/fexolm/git/SLama/tests/matrix-base-algo-test.c:56]: (style) Variable 'matrix1' is reassigned a value before the old one has been used.
[/home/fexolm/git/SLama/tests/matrix-base-algo-test.c:86] -> [/home/fexolm/git/SLama/tests/matrix-base-algo-test.c:87]: (style) Variable 'matrix' is reassigned a value before the old one has been used.
[/home/fexolm/git/SLama/tests/matmul-test.c:15] -> [/home/fexolm/git/SLama/tests/matmul-test.c:19]: (style) Variable 'matrix1' is reassigned a value before the old one has been used.
[/home/fexolm/git/SLama/tests/matmul-test.c:16] -> [/home/fexolm/git/SLama/tests/matmul-test.c:20]: (style) Variable 'matrix2' is reassigned a value before the old one has been used.
[/home/fexolm/git/SLama/tests/matmul-test.c:17] -> [/home/fexolm/git/SLama/tests/matmul-test.c:56]: (style) Variable 'result' is reassigned a value before the old one has been used.
[/home/fexolm/git/SLama/tests/base-matrix-test.c:15] -> [/home/fexolm/git/SLama/tests/base-matrix-test.c:16]: (style) Variable 'matrix' is reassigned a value before the old one has been used.
[/home/fexolm/git/SLama/tests/base-matrix-test.c:27] -> [/home/fexolm/git/SLama/tests/base-matrix-test.c:29]: (style) Variable 'matrix' is reassigned a value before the old one has been used.
[/home/fexolm/git/SLama/tests/base-matrix-test.c:28] -> [/home/fexolm/git/SLama/tests/base-matrix-test.c:35]: (style) Variable 'matrix1' is reassigned a value before the old one has been used.
[/home/fexolm/git/SLama/tests/base-matrix-test.c:46] -> [/home/fexolm/git/SLama/tests/base-matrix-test.c:48]: (style) Variable 'matrix' is reassigned a value before the old one has been used.
[/home/fexolm/git/SLama/tests/base-matrix-test.c:47] -> [/home/fexolm/git/SLama/tests/base-matrix-test.c:49]: (style) Variable 'matrix1' is reassigned a value before the old one has been used.
[/home/fexolm/git/SLama/tests/base-matrix-test.c:65] -> [/home/fexolm/git/SLama/tests/base-matrix-test.c:66]: (style) Variable 'matrix' is reassigned a value before the old one has been used.
[/home/fexolm/git/SLama/tests/base-matrix-test.c:83] -> [/home/fexolm/git/SLama/tests/base-matrix-test.c:86]: (style) Variable 'matrix' is reassigned a value before the old one has been used.
[/home/fexolm/git/SLama/tests/base-matrix-test.c:84] -> [/home/fexolm/git/SLama/tests/base-matrix-test.c:90]: (style) Variable 'matrix1' is reassigned a value before the old one has been used.
(information) Cppcheck cannot find all the include files (use --check-config for details)

**Review by fexolm — CHANGES_REQUESTED** [review]

File: HaroldAdviser/Controllers/PipelineController.cs (hunk 1 above)
Comment (weeellz) [reply HaroldAdviser/Controllers/PipelineController.cs]: done

File: HaroldAdviser/Controllers/UserController.cs (hunk 2 above)
Comment (weeellz) [reply HaroldAdviser/Controllers/UserController.cs:122]: done

File: HaroldAdviser/Views/User/AllRepositories.cshtml [hunk 3]
```diff
@@ -11,4 +11,5 @@
 @section scripts
     {
     <script src="~/Scripts/repository-check.js"></script>
+    @*<script src="~/Scripts/get-warnings.js"></script>*@
```
Comment (fexolm) [inline HaroldAdviser/Views/User/AllRepositories.cshtml]: Please remove comment. Also, this script is for repository view page. Please move it here.

File: HaroldAdviser/wwwroot/Scripts/get-warnings.js [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+
+var get_warnings = function() {
+    $.get("/api/user/repository/warnings",
+        function(data) {
+            $.each(data,
+                function(ind, val) {
+
+                    //TODO: add class for warnings
+
+                    $("").append(val.kind + val.file + val.lines + val.message);
```
Comment (fexolm) [inline HaroldAdviser/wwwroot/Scripts/get-warnings.js]: Please add this class in repository view page.

**Review by fexolm — CHANGES_REQUESTED** [review]

File: HaroldAdviser/Views/User/AllRepositories.cshtml (hunk 3 above)
Comment (weeellz) [reply HaroldAdviser/Views/User/AllRepositories.cshtml]: done

File: HaroldAdviser/wwwroot/Scripts/get-warnings.js (hunk 4 above)
Comment (weeellz) [reply HaroldAdviser/wwwroot/Scripts/get-warnings.js]: done

**Review by fexolm — APPROVED** [review]

**Pull request merged**